### ISSUE TITLE
[data grid] `getVisibleRowModels` gets all rows

### ISSUE THREAD

**Issue by JvilleDjango** [mui/mui-x#3021]
- [ x] The issue is present in the latest release.
- [x ] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

getVisibleRowModels() gets all rows in the table

## Expected Behavior 🤔

getVisibleRowModels() according to the documentation is should only get the visible rows

## Steps to Reproduce 🕹

Steps:

1. import { useGridApiRef, DataGridPro } from "@mui/x-data-grid-pro";
2. const apiRef = useGridApiRef();
3. <DataGridPro apiRef={apiRef} ... /> add data of course
4. add an event and log out apiRef.current.getVisileRowModels()

## Your Environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>
  
```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by flaviendelangle** [comment]
`5.2.3` is the version of the core packages (`@mui/material` for instance).
The X packages (`@mui/x-data-grid-pro` for instance) are on version `5.2.0` right now.

But no these two selectors are still exported.

**Comment by JvilleDjango** [comment]
That's right, I was thinking I was pulling those two from the api. my bad. 
 
 I'm noticing an issue that was in version 4 that got fixed but, looks like it may have returned. 
 onPageSizeChange() is running on mount and it changing my set rows causing the scrollbar to not show.  I'm setting the rows to 100 initially but, when it mounts its setting it to 11. This worked before but, not anymore and the main reason for the above question.

**Comment by flaviendelangle** [comment]
Could you give me a reproduction case ? [This template](https://codesandbox.io/s/datagridpro-v5-quick-start-5pjhc) should be a good starting point.

**Comment by JvilleDjango** [comment]
Actually I found the issue. I had added "autoPageSize" to the table. Once I removed it. It worked.

**Comment by igorSurz** [comment]
can someone help here, useMemo doesnt helps. according to documentation apiref object will be populated after first render but it still not works with this `gridPaginatedVisibleSortedGridRowIdsSelector` function

here is the code [https://codesandbox.io/s/datagridpro-v5-quick-start-5pjhc?file=/src/App.tsx](url)

**Comment by igorSurz** [comment]
Dear smart people, can you please help me with this, burned half of the day and I just don't get it what is the problem with `gridPaginatedVisibleSortedGridRowIdsSelector `

here is the code https://codesandbox.io/s/data-grid-pro-forked-etfdb3?file=/src/App.tsx

**Comment by shoxter** [comment]
> The name of this method is wrong. What it returns is not the visible rows but the filtered rows. If pagination is on, changing the page and calling this method will return the same thing. I don't think we need a reproduction.
> 
> https://github.com/mui-org/material-ui-x/blob/adc0468e18044f34cdffe7a084cfdab2f45bab62/packages/grid/_modules_/grid/models/api/gridFilterApi.ts#L44
> 
> +1 for renaming it to `getFilteredRows` as well as the related selectors. To get the real visible rows, we need to expose the rows used on virtualization:
> 
> https://github.com/mui-org/material-ui-x/blob/adc0468e18044f34cdffe7a084cfdab2f45bab62/packages/grid/_modules_/grid/components/GridVirtualScroller.tsx#L129

+1 here. "Visible" means what is currently in the viewport or what is currently in the DOM based on virtualization. Glad to come across this for clarification.

**Comment by michelengelen** [comment]
The confusion here was a naming problem, and it was resolved by the v6 rename: `getVisibleRowModels` no longer exists. The concepts are now separate selectors:

```tsx
const filteredRows = gridExpandedSortedRowEntriesSelector(apiRef);
const pageRows = gridPaginatedVisibleSortedGridRowEntriesSelector(apiRef);
```

The first returns rows that passed filtering, the second returns exactly the rows on the current page, which is what was asked in the thread. See https://mui.com/x/react-data-grid/state/#access-the-state.

Closing as completed. If a selector returns unexpected rows on a current version, please open a new issue with a reproduction.
